modules/contact/src/constraints/ExplicitDynamicsContactConstraint.C

```diff
@@ -430,25 +368,30 @@ ExplicitDynamicsContactConstraint::gapOffset(const Node & node)
 }
 
 Real
-ExplicitDynamicsContactConstraint::nodalArea(const Node & node)
+ExplicitDynamicsContactConstraint::getPenalty(const Node & /*node*/)
 {
-  dof_id_type dof = node.dof_number(_aux_system.number(), _nodal_area_var->number(), 0);
-
-  Real area = (*_aux_solution)(dof);
-  if (area == 0.0)
-  {
-    if (_t_step > 1)
-      mooseError("Zero nodal area found");
-    else
-      area = 1.0; // Avoid divide by zero during initialization
-  }
-
-  return area;
+  // TODO: Include normalized penalty values.
+  return _penalty;
 }
 
 Real
-ExplicitDynamicsContactConstraint::getPenalty(const Node & /*node*/)
+ExplicitDynamicsContactConstraint::computeFaceMass(const NumericVector<Real> & lumped_mass)
 {
-  // TODO: Include normalized penalty values.
-  return _penalty;
+  // Initialize face mass to zero
+  Real mass_face(0.0);
+
+  // Get the primary side of the current contact
+  const auto primary_side = _current_primary->side_ptr(_assembly.neighborSide());
+
+  // Get the dofs on the primary (face) side of the contact
+  std::vector<dof_id_type> face_dofs;
+  _primary_var.getDofIndices(primary_side.get(), face_dofs);
```

Comment:
do we have assertions somewhere that this constraint can only be used with H1 finite element families?

Reply:
Would this specifically fail with other finite elemet families?

Reply:
it would not work with discontinuous families since the dofs are all associated with the element and not with the nodes which compose the side pointer element you build

Reply:
Looks like NodeFaceConstraint will error out if the primary variable is not "nodal", when it uses the coupleable interface.